src/Compilers/CSharp/Test/Semantic/Semantics/NullableReferenceTypesTests.cs

```diff
@@ -82658,9 +82752,9 @@ static void F3((B?, B) t)
             var comp = CreateCompilation(source, options: WithNonNullTypesTrue());
             // https://github.com/dotnet/roslyn/issues/32599: Handle tuple element conversions.
             comp.VerifyDiagnostics(
-                // (16,12): warning CS8620: Argument of type '(A?, A)?' cannot be used for parameter 't' of type '(A, A?)?' in 'void Program.F2((A, A?)? t)' due to differences in the nullability of reference types.
+                // (16,12): warning CS8620: Argument of type 'S' cannot be used for parameter 't' of type '(A, A?)?' in 'void Program.F2((A, A?)? t)' due to differences in the nullability of reference types.
```

Comment:
This doesn't look right at a glance. The diagnostic doesn't mention the conversion used to even compare the `S` to the `(A, A?)?`, so it feels like perhaps a downgrade in diagnostic quality.

Not sure though. It definitely doesn't look like a very common scenario.

Reply:
This is a known issue ([https://github.com/dotnet/roslyn/issues/31798](https://github.com/dotnet/roslyn/issues/31798)). Added link for tracking

---
In reply to: [522506769](https://github.com/dotnet/roslyn/pull/49279#discussion_r522506769) [](ancestors = 522506769)